components/merkledb/src/access/extensions.rs

```diff
@@ -23,13 +23,14 @@ use crate::{
 };
 
 /// Extension trait allowing for easy access to indexes from any type implementing
-/// `Access`.
+/// [`Access`] + `Copy`.
```

Comment:
Explanation why both extension traits are necessary could be helpful. Something like

> This trait is helpful for *references* implementing `Access`, such as `&Fork` or `&dyn Snapshot` because Rust method resolution does not apply `AccessExt` to variables of corresponding types. For example, if `fork` has type `Fork`, then `fork.get_list("foo")` is not resolved as `AccessExt::get_list(..)`, only `(&fork).get_list("foo")` is.

(This assumes that the traits are renamed as mentioned in another comment.)

Reply:
I'd also expect cross-references between traits in their descriptions, like 

```
For a version on `AccessExt` traits designed for `Copy` types (e.g. `&Fork` and
`&dyn Snapshot`) see [`CopyAccessExt` trait](trait.CopyAccessExt.html).
```